tests/kubernetes/application/test_controller_wrapper.py

```diff
@@ -167,74 +167,95 @@ def test_sync_horizontal_pod_autoscaler_no_autoscaling():
     # Do nothing
     config_dict = {"instances": 1}
     app = setup_app(config_dict, False)
-    app.sync_horizontal_pod_autoscaler(kube_client=mock_client)
```

Comment:
You should write tests for creating and deleting the HPA based solely on whether or not the autoscaler is paused.

(This comment isn't related to this line, I just needed it to appear as a standalone comment)